File: src/utils/logger.ts
import pino, { Logger, LoggerOptions } from 'pino';
import pinoPretty from 'pino-pretty';
import fs from 'fs';
import path from 'path';
import { CONFIG } from '../config/settings';

const baseOptions: LoggerOptions = {
  name: 'DeepCurrents',
  level: CONFIG.LOG_LEVEL,
  base: undefined,
  timestamp: pino.stdTimeFunctions.isoTime,
};

const streams: Array<{ stream: any }> = [];

if (CONFIG.LOG_TO_STDERR) {
  if (CONFIG.LOG_PRETTY) {
    streams.push({
      stream: pinoPretty({
        colorize: true,
        translateTime: 'SYS:standard',
        destination: 2,
      }),
    });
  } else {
    streams.push({ stream: pino.destination(2) });
  }
}

if (CONFIG.LOG_TO_FILE) {
  const resolvedPath = path.resolve(process.cwd(), CONFIG.LOG_FILE_PATH);
  fs.mkdirSync(path.dirname(resolvedPath), { recursive: true });
  streams.push({ stream: pino.destination(resolvedPath) });
}

if (streams.length === 0) {
  streams.push({ stream: pino.destination(2) });
}

const rootLogger: Logger = pino(baseOptions, pino.multistream(streams));

export function getLogger(component: string): Logger {
  return rootLogger.child({ component });
}

export { rootLogger };

File: src/utils/logger.py
from loguru import logger
import sys
import os
from ..config.settings import CONFIG

def setup_logger():
    # 移除默认处理器
    logger.remove()

    # 配置标准错误输出
    if CONFIG.log_to_stderr:
        logger.add(
            sys.stderr,
            level=CONFIG.log_level,
            colorize=CONFIG.log_pretty,
            format="<green>{time:YYYY-MM-DD HH:mm:ss.SSS}</green> | <level>{level: <8}</level> | <cyan>{name}</cyan>:<cyan>{function}</cyan>:<cyan>{line}</cyan> - <level>{message}</level>"
        )

    # 配置本地文件落盘
    if CONFIG.log_to_file:
        log_dir = os.path.dirname(CONFIG.log_file_path)
        if log_dir and not os.path.exists(log_dir):
            os.makedirs(log_dir)
            
        logger.add(
            CONFIG.log_file_path,
            level=CONFIG.log_level,
            rotation="10 MB",
            retention="7 days",
            format="{time:YYYY-MM-DD HH:mm:ss.SSS} | {level: <8} | {name}:{function}:{line} - {message}"
        )

def get_logger(name: str):
    return logger.bind(name=name)

# 初始化全局日志配置
setup_logger()
